phpBB/includes/acp/acp_search.php

```diff
@@ -122,23 +177,23 @@ function settings($id, $mode)
 			// e.g. integer:4:12 (min 4, max 12)
 			$var_type = explode(':', $var_type);
 
-			$config_value = $cfg_array[$config_name];
-			settype($config_value, $var_type[0]);
+			$this->config_value = $cfg_array[$config_name];
```

Comment:
`$this->config_value` is not properly defined. I think you changed this by accident when changing `$config` to `$this->config`